sdk/core/core-lro/src/lroEngine/azureAsyncPolling.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  failureStates,
+  LroResourceLocationConfig,
+  LongRunningOperation,
+  LroBody,
+  LroResponse,
+  LroStatus,
+  RawResponse,
+  successStates
+} from "./models";
+import { isUnexpectedPollingResponse } from "./requestUtils";
+
+function getResponseStatus(rawResponse: RawResponse): string {
+  const { status } = (rawResponse.body as LroBody) ?? {};
+  return status?.toLowerCase() ?? "succeeded";
+}
+
+function isAzureAsyncPollingDone(rawResponse: RawResponse): boolean {
+  const state = getResponseStatus(rawResponse);
+  if (isUnexpectedPollingResponse(rawResponse) || failureStates.includes(state)) {
+    throw new Error(`Operation status: ${state}`);
+  }
+  return successStates.includes(state);
+}
+
+async function sendFinalRequest<TResult>(
```

Comment:
Please document these functions! Someone new learning this could have a hard time understanding the purpose of these.

Reply:
Please document all the internal functions that aren’t self-evident, if that makes sense

Reply:
Done.